pkg/workflows/builder.go

```diff
@@ -0,0 +1,73 @@
+package workflows
+
+import (
+	"fmt"
+	"strconv"
+)
+
+// 1. Capability defines JSON schema for inputs and outputs of a capability.
+// Trigger: triggerOutputType := workflowBuilder.addTrigger(DataStreamsTrigger.Config{})
+// Adds metadata to the builder. Returns output type.
+// 2. Consensus: consensusOutputType := workflowBuilder.addConsensus(ConsensusConfig{
+// 	Inputs: triggerOutputType,
+// })
+
+type Workflow struct {
+	spec *workflowSpecYaml
+}
+
+type Trigger[O any] struct {
+	Definition TriggerDefinitionYaml
+	Output     O
+}
+
+type Consensus[O any] struct {
+	Definition StepDefinitionYaml
+	Output     O
+}
+
+type CapabilityDefinition[O any] struct {
+	Ref    string
+	Output O
+}
+
+type NewWorkflowParams struct {
+	Owner string
+	Name  string
+}
+
+func NewWorkflow(
+	params NewWorkflowParams,
+) *Workflow {
+	return &Workflow{
+		spec: &workflowSpecYaml{
+			Owner: params.Owner,
+			Name:  params.Name,
+		},
+	}
+}
+
+func AddTrigger[O any](w *Workflow, trigger Trigger[O]) CapabilityDefinition[O] {
+	// Add ref to trigger.Definition
+	trigger.Definition.Ref = fmt.Sprintf("trigger-%s", strconv.Itoa((len(w.spec.Triggers))))
```

Comment:
The generation of the ref is confusing.  It disallows users from having names that make sense to them.  If I ever add/remove a step I can't easily consolidate logs with prior runs.